### PR TITLE
feature-#164047025-User should be able to login with Fa…

### PR DESCRIPTION
**Pull request by kelvinndmo**
## Description ##
This setups up social authentication with Google and Facebook to the application

## Type of change ##
Please select the relevant option
- [ ] Bug fix(a non-breaking change which fixes an issue)
- [x] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested? ##
Please describe the tests that you ran to verify your changes
- [ ] End to end test
- [ ] Integration test
- [x] unit test

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## Related Pivotal Tracker stories ##
[#164047025](https://www.pivotaltracker.com/story/show/164047025)

## Reviewers ##
- @andela/olympian-devs
- @Derrickkip 

#Screenshots

Screen Shot 2019-04-16 at 4 10 41 PM

### CODE REVIEW

**Review by mnswaleh — COMMENTED** [review]
place the social login buttons on the login page

**Comment by kelvinndmo** [comment]
> place the social login buttons on the login page

@mnswaleh done.Thanks for the review

File: package.json [hunk 1]
```diff
@@ -35,7 +46,7 @@
     "watch-css": "npm run build-css && node-sass --include-path ./src --include-path ./node_modules src/ -o src/ --watch --recursive",
     "start": "npm-run-all -p watch-css start-js",
     "build": "npm run build-css && react-scripts build",
-    "start-js": "react-scripts start",
+    "start-js": "PORT=4000 react-scripts start",
```
Comment (Kibetchirchir) [inline package.json]: remove the port number please

File: src/components/Home/__snapshots__/App.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (kevpy) [inline src/components/Home/__snapshots__/App.test.js.snap]: This file should not be tracked. Please remove it.

File: src/components/Home/Home.css [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.App-header {
```
Comment (kevpy) [inline src/components/Home/Home.css]: Please avoid tracking `.css` files. We are using `Sass`. In this context please remove it form files being tracked by git

**Review by Kibetchirchir — CHANGES_REQUESTED** [review]

File: src/components/social/styles.css [hunk 4]
```diff
@@ -0,0 +1,53 @@
+.novak {
```
Comment (kevpy) [inline src/components/social/styles.css]: Also, add this file to the file that need not be tracked

File: src/components/Login/Login.test.js [hunk 5]
```diff
@@ -1,98 +1,92 @@
...
-import { combineReducers, applyMiddleware, createStore } from 'redux';
-import { Provider } from 'react-redux';
-import thunk from 'redux-thunk';
-import Enzyme, { shallow, mount } from 'enzyme';
-import Adapter from 'enzyme-adapter-react-16';
-import loginReducer from '../../reducers/index';
-import loginAction from '../../actions/index';
-import { Login } from './Login';
-import store from '../../store/store';
+import React from 'react'
```
Comment (kevpy) [inline src/components/Login/Login.test.js]: Please undo the formatting of the code added. Absence of trailing commas and semi-colons will cause hound to flag the current PR and any other PR that contains the same file as not meeting `eslint` rules. This applies to the `Login.js` file as well

Comment (kelvinndmo) [reply src/components/Login/Login.test.js]: noted

File: src/components/social/styles.css (hunk 4 above)
Comment (kelvinndmo) [reply src/components/social/styles.css]: noted working on it

File: src/components/Login/Login.css [hunk 6]
```diff
@@ -0,0 +1,99 @@
+body {
```
Comment (kevpy) [inline src/components/Login/Login.css]: Please remove this file. It shouldn't be tracked.

File: src/actions/auth.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
+// import axios from 'axios';
```
Comment (kevpy) [inline src/actions/auth.js]: If this file isn't needed in your PR, please remove it. It is commented out, thus this means it is not needed.

**Review by Kibetchirchir — APPROVED** [review]

**Review by kevpy — APPROVED** [review]

File: src/components/Login/Login.js [hunk 8]
```diff
@@ -135,9 +136,9 @@ export class Login extends Component {
                   noValidate
                   onChange={this.handleChange}
                 />
-                {/*{formErrors.password.length > 0 && (*/}
-                {/*  <span className="errorMessage">{formErrors.password}</span>*/}
-                {/*)}*/}
+                {/* {formErrors.password.length > 0 && ( */}
+                {/*  <span className="errorMessage">{formErrors.password}</span> */}
+                {/* )} */}
```
Comment (Derrickkip) [inline src/components/Login/Login.js]: Remove the commented lines

File: src/components/social/index.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    this.setState({
+      error: 'Something went wrong',
+    });
+  };
+
+  renderLoginButton = (provider, appId, content) => {
+    return (
+      <SocialButton
+        className={`waves-effect waves-light btn-flat btn--default m-b--15 button social-button ${provider}`}
+        id="#facebook-connect"
```
Comment (Derrickkip) [inline src/components/social/index.js]: Having this id Here causes there to be two buttons with the same ID, you should pass it as props

File: src/components/social/index.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+  }
+}
+
+SocialAuth.propTypes = {
+  socialogin: PropTypes.func.isRequired,
+  user: PropTypes.object.isRequired,
+};
+
+SocialAuth.propTypes = {
+  history: PropTypes.shape({
```
Comment (Derrickkip) [inline src/components/social/index.js:93]: Why do you have two descriptions of the propTypes?

**Review by Derrickkip — APPROVED** [review]

**Pull request merged**